src/uct/sm/cma/cma_ep.c

```diff
@@ -0,0 +1,76 @@
+/**
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
+* $COPYRIGHT$
+* $HEADER$
+*/
+
+#define _GNU_SOURCE
```

Comment:
IMO we should not define _GNU_SOURCE directly but use #include <features.h>
configure.ac checks for it and it contains all platform dependent features including GNU_SOURCE.